smoke-test/src/test/java/org/opennms/smoketest/OpenNMSCloudPluginIT.java

```diff
@@ -0,0 +1,161 @@
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opennms.smoketest.stacks.OpenNMSStack;
+import org.opennms.smoketest.stacks.StackModel;
+import org.opennms.smoketest.utils.KarafShellUtils;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+
+public class OpenNMSCloudPluginIT {
+
+    private static final String PATH_TO_FILE = "target/plugin-cloud-assembly/org.opennms.plugins.cloud-assembly.kar";
+    private static final String PATH_TO_CLOUD_MOCK = "target/cloud-mock-with-dependencies/org.opennms.plugins.cloud-mock-with-dependencies.jar";
+    private static final String CONTAINER_PATH = "/usr/share/opennms/deploy/";
+    private static final String CLOUD_MOCK_JAR_NAME = "org.opennms.plugins.cloud-mock-with-dependencies.jar";
+    private static final String CLOUD_MOCK_MAIN = "org.opennms.plugins.cloud.ittest.MockCloudMain";
+    private static final String KAR_FILE_NAME = "org.opennms.plugins.cloud-assembly.kar";
+
+    @ClassRule
+    public static OpenNMSStack stack = OpenNMSStack
+            .withModel(StackModel.newBuilder()
+                    .withSentinel()
+                    .build());
+
+    @ClassRule
+    public static GenericContainer mockCloudContainer = new GenericContainer(DockerImageName.parse("opennms/deploy-base:jre-2.0.6.b165"))
+            .withCreateContainerCmdModifier(cmd -> {
+                final CreateContainerCmd createCmd = (CreateContainerCmd) cmd;
+                createCmd.withEntrypoint("/bin/bash");
+                createCmd.withName("cloudMock");
+            }).withCopyFileToContainer(MountableFile.forHostPath(PATH_TO_CLOUD_MOCK), format("%s%s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME))
+            .withCommand("-c", format("java -cp %s%s %s", CONTAINER_PATH, CLOUD_MOCK_JAR_NAME, CLOUD_MOCK_MAIN)).withExposedPorts(9003)
+            .withNetwork(stack.opennms().getNetwork())
+            .withNetworkAliases("cloudMock");
+
+    @Before
+    public void setUp() {
+        mockCloudContainer.start();
+    }
+
+    @Test
+    public void installCloudPlugin() throws Exception {
+        installCloudPluginInOpennmsMustBeSuccessful();
+        installCloudPluginInSentinelMustBeSuccessful();
+    }
+
+    public void installCloudPluginInOpennmsMustBeSuccessful() throws Exception {
+        // Given
+        stack.opennms().copyFileToContainer(MountableFile.forHostPath(PATH_TO_FILE), format("%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        String config = createConfig();
+        await().atMost(20, SECONDS);
```

Comment:
What are we waiting for? Try to avoid unconditional awaits / sleeps by testing for some kind of condition which signals readiness.

Reply:
Unconditional waiting has two drawbacks here:
- CI pipeline is billed by time
- The will flap if CI is having a slow day

Reply:
this can be probably removed